services/src/main/java/org/keycloak/services/resources/account/AccountCredentialResource.java

```diff
@@ -57,6 +57,139 @@ public AccountCredentialResource(KeycloakSession session, EventBuilder event, Us
 //        models.forEach(c -> c.setSecretData(null));
 //        return models.stream().map(ModelToRepresentation::toRepresentation).collect(Collectors.toList());
 //    }
+    
+    private static class CredentialContainer {
+        private String category;
+        private String type;
+        private String tooltip;
+        private boolean enabled;
+        private String createAction;
+        private String updateAction;
+        private boolean removeable;
+        private List<CredentialModel> userCredentials;
+        
+        public CredentialContainer(String category, String type, String tooltip, boolean enabled, String createAction, String updateAction, boolean removeable,List<CredentialModel> userCredentials) {
+            this.category = category;
+            this.type = type;
+            this.tooltip = tooltip;
+            this.enabled = enabled;
+            this.createAction = createAction;
+            this.updateAction = updateAction;
+            this.removeable = removeable;
+            this.userCredentials = userCredentials;
+        }
+
+        public String getCategory() {
+            return category;
+        }
+        
+        public String getType() {
+            return type;
+        }
+
+        public String getTooltip() {
+            return tooltip;
+        }
+
+        public boolean isEnabled() {
+            return enabled;
+        }
+
+        public String getCreateAction() {
+            return createAction;
+        }
+
+        public String getUpdateAction() {
+            return updateAction;
+        }
+
+        public boolean isRemoveable() {
+            return removeable;
+        }
+
+        public List<CredentialModel> getUserCredentials() {
+            return userCredentials;
+        }
+        
+    }
+    
+    @GET
+    @NoCache
+    @Produces(javax.ws.rs.core.MediaType.APPLICATION_JSON)
+    public List<CredentialContainer> dummyCredentialTypes(){
+        auth.requireOneOf(AccountRoles.MANAGE_ACCOUNT, AccountRoles.VIEW_PROFILE);
+        List<CredentialModel> models = session.userCredentialManager().getStoredCredentials(realm, user);
+        
+        List<CredentialModel> passwordUserCreds = new java.util.ArrayList<>();
+        passwordUserCreds.add(models.get(0));
```

Comment:
This will throw an exception in case that user doesn't have password? Also in case he doesn't have password and has 1 OTP credential, it can happen that "models.get(0)" won't return the password.

I know we're going to rewrite this stuff and it is just "dummy" endpoint for now. So leaving to you whether to change this or not. I am fine with accept PR even without fixing it we all agree that some temporary "bugs" in this dummy endpoints are ok.

Reply:
@mposolda I am assuming that you will take care of those things when there is live data.  As long as the data you send is in the format I expect, the UI will be able to handle it.

I am only assuming that there will be at least one credential sent to me.  It doesn't matter what category or type it is.

Reply:
> > I just realized the tooltips are missing.
> 
> Actually need to discuss this a bit. Right now, tooltips are sent for each credential type. Should they also appear for each credential category?

@mposolda @abstractj I think I will rename "tooltip" to "helptext".  It will be shown as plain text after the type name instead of as a tooltip.

Concerning help text at the category level, with the current structure we can not have help text that belongs to the category.  This is because you can have more than one type for each category and they are sent as a category/type combo.  We would need another level of abstraction to send an attribute that belongs specifically to the category.

Reply:
@ssilvert whatever you and @mposolda are in agreement, that's perfectly fine from me. Apicurio was great to be used as a starting point, but you guys doing the implementation know best what has to be done.

Reply:
@ssilvert The UI with added helpText for each credentialType looks nice to me. Thanks for adding it!

Regarding the helpText for the category, I think it will makes sense to have helpText which will be specific to each category. I just don't know where exactly to show it in the UI, so that it looks nice and user-friendly...

Regarding your question for helpText on categories, there will be always just 3 categories. Those 3 categories are "hardcoded" (with the exception that they should be localized, but otherwise it is pretty much hardcoded text) and their helpTexts can be hardcoded as well. There is no need to return anything from the REST API regarding helpText for the categories (which I am not sure if it is something you meant in your last paragraph).

Reply:
@mposolda OK.  I'll hard-code some text then.  I think I'll post to the dev list to get feedback on all the help text.